sae_lens/training/training_sae.py

```diff
@@ -139,6 +143,15 @@ def __init__(self, cfg: TrainingSAEConfig, use_error_term: bool = False):
         base_sae_cfg = SAEConfig.from_dict(cfg.get_base_sae_cfg_dict())
         super().__init__(base_sae_cfg)
         self.cfg = cfg  # type: ignore
+
+        self.encode_with_hidden_pre_fn = (
+            self.encode_with_hidden_pre
+            if cfg.architecture != "gated"
+            else self.encode_with_hidden_pre_gated
+        )
+
+        self.check_cfg_compatibility()
```

Comment:
We're checking config compatibility elsewhere (in the training config init). I think I'd prefer to put this there.